service-workers/service-worker/navigation-preload/broken-chunked-encoding.https.html

```diff
@@ -5,38 +5,48 @@
 <script src="/resources/testharnessreport.js"></script>
 <script src="../resources/test-helpers.sub.js"></script>
 <script>
-promise_test(t => {
-    var script = 'resources/broken-chunked-encoding-worker.js';
-    var scope = 'resources/broken-chunked-encoding-scope.asis';
-    return service_worker_unregister_and_register(t, script, scope)
-      .then(registration => {
-          add_completion_callback(_ => registration.unregister());
-          var worker = registration.installing;
-          return wait_for_state(t, worker, 'activated');
-        })
-      .then(_ => with_iframe(scope))
-      .then(frame => {
-          assert_equals(
-            frame.contentDocument.body.textContent,
-            'PASS: preloadResponse resolved');
-        });
-  }, 'FetchEvent#preloadResponse resolves even if the body is sent with broken chunked encoding.');
-
-promise_test(t => {
-    var script = 'resources/broken-chunked-encoding-worker.js';
-    var scope = 'resources/chunked-encoding-scope.py?use_broken_body';
-    return service_worker_unregister_and_register(t, script, scope)
-      .then(registration => {
-          add_completion_callback(_ => registration.unregister());
-          var worker = registration.installing;
-          return wait_for_state(t, worker, 'activated');
-        })
-      .then(_ => with_iframe(scope))
-      .then(frame => {
-          assert_equals(
-            frame.contentDocument.body.textContent,
-            'PASS: preloadResponse resolved');
-        });
-  }, 'FetchEvent#preloadResponse resolves even if the body is sent with broken chunked encoding with some delays');
+promise_test(async t => {
+  const SCRIPT = 'resources/broken-chunked-encoding-worker.js';
+  const SCOPE = 'resources/broken-chunked-encoding-scope.asis';
+
+  const registration =
+      await service_worker_unregister_and_register(t, SCRIPT, SCOPE);
+  const worker = registration.installing;
+  await wait_for_state(t, worker, 'activated');
+  const frame = await with_iframe(SCOPE);
+
+  add_completion_callback(async() => {
+    if (registration)
+      await registration.unregister();
+    if (frame)
+      frame.remove();
```

Comment:
It's preferable to remove the frame before unregistering, since unregister doesn't actually remove the registration until the frames are closed. (same comment throughout)

Reply:
Updated the code in latest commit.